Assets/Scripts/Player/Controller.cs

```diff
@@ -290,5 +290,44 @@ private void OnTriggerExit2D(Collider2D other)
         }
 
         #endregion
+        
+        #region Public variables
+        
+        [Range(0, 10)]
+        public int hitPoints = 10;
+        
+        /// <summary>
+        /// Events listened by the HUD to update the HUD health bar.
+        /// </summary>
+        public event Action UpdateHealthBarHUD;
+        public event Action ResetHealthBarHUD;
```

Comment:
I would make the type of these events Action<int> so that each time they're called, you can pass in the player's updated hit points

Reply:
Also, if there are no subscribers to these events, invoking them will cause a null pointer exception.
Can you change the way you call them to `EventName?.Invoke(arg0, arg1...)`?

Reply:
done